File: LeetCode/Medium/0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1;
        int currAmount = 0;
        while (left < right) {
            int w = right - left;
            int h = Math.min(height[right], height[left]);
            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
            currAmount = Math.max(currAmount, (h * w));
        }
        return currAmount;
    }
}

File: LeetCode/Medium/0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
    let left = 0, right = height.length - 1
    let maxArea = 0
    while (left < right) {
        let currArea = Math.min(height[right], height[left]) * (right - left)
        maxArea = Math.max(maxArea, currArea)
        if (height[right] > height[left]) {
            left++
        } else {
            right--
        }
    }
    return maxArea
};